internal/database/testing/dqlitesuite.go

```diff
@@ -214,15 +197,17 @@ func (s *DqliteSuite) OpenDBForNamespace(c *tc.C, domain string, foreignKey bool
 	c.Assert(err, tc.ErrorIsNil)
 
 	// Ensure we close all databases that are opened during the tests.
-	s.cleanupDB(domain, db)
+	c.Cleanup(func() {
+		err := db.Close()
+		c.Check(err, tc.ErrorIsNil)
+	})
```

Comment:
This is excellent. I wanted to do this from the get-go, but didn't have a nice way to do it. Now we do!